crates/type-system/src/ontology/uri/mod.rs

```diff
@@ -1,18 +1,20 @@
 mod error;
 #[cfg(target_arch = "wasm32")]
 mod wasm;
-
 use std::{fmt, result::Result, str::FromStr, sync::LazyLock};
 
 use error::ParseVersionedUriError;
 use regex::Regex;
 use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+#[cfg(target_arch = "wasm32")]
+use tsify::Tsify;
 use url::Url;
 
 use crate::uri::error::ParseBaseUriError;
 
+#[cfg_attr(target_arch = "wasm32", derive(Tsify))]
 #[derive(Clone, PartialEq, Eq, Hash)]
-pub struct BaseUri(Url);
+pub struct BaseUri(#[cfg_attr(target_arch = "wasm32", tsify(type = "string"))] Url);
```

Comment:
To make this more readable we probably want to convert it into a `struct { ... }` (instead of `struct(...)`

Reply:
I do wish there was a better way to inject this attr :(

I hear you, but I don't think we should swap to a struct instead of a newtype just for readability of its initial definition. Also having it as a newtype is somewhat important for the `tsify` as the generated Typescript type is an alias rather than a class/interface.

Reply:
That makes sense, thanks for clarifying!